src/worker/plugins/teardown.ts

```diff
@@ -1,11 +1,16 @@
 import { processError } from '../../shared/error'
-import { PluginsServer } from '../../types'
+import { PluginConfig, PluginsServer } from '../../types'
 
-export async function teardownPlugins(server: PluginsServer): Promise<void> {
-    const { pluginConfigs } = server
+export async function teardownPlugins(
+    server: PluginsServer,
+    pluginConfigs: PluginConfig[] | undefined = undefined
+): Promise<void> {
+    if (!pluginConfigs) {
+        pluginConfigs = Array.from(server.pluginConfigs.values())
+    }
```

Comment:
Why not deconstruct in loop like before?

Reply:
I cleaned it up a bit by changing the function's argument from an array to a single `PluginConfig`. The `Array.from()` code from before was to make TS happy. With the new approach it's not needed.